### PR TITLE
docs: fix admission control sample usage

### PR DESCRIPTION
**Pull request by hshamji**
Commit Message: docs: fix admission control sample usage
Docs Changes: Please see [discussion in Issue here](https://github.com/envoyproxy/envoy/issues/21221)

### CODE REVIEW

**Review by daixiang0 — DISMISSED** [review]
Good catch, LGTM.

@phlax should we also backport this? Since we have changed it a few versions ago.

**Review by phlax — DISMISSED** [review]
@hshamji thanks for following up on this

im happy to land this as is, but with these bad configs in the docs i encourage moving them to yaml files so they are tested in CI

if you were up for doing that i can wait, otherwise i can follow up myself

**Comment by phlax** [comment]
> should we also backport this?

@daixiang0  makes sense altho probably not very high priority atm - i would be happy to review backport PRs once this one has landed

**Comment by phlax** [comment]
> do you have any pointers for where I would make this change? 

yep if you:

```console
# git grep literalinclude docs/

```

you should see where existing configs are included - the config file will need to be a full "bootstrap" config and will be automatically validated and tested

**Comment by hshamji** [comment]
@phlax , I've moved the config out to another file and fixed a few things in the docs.

I tried adding some garbled text to the config to see whether the CI would reject the config as invalid, but it still passed. I tried that with this newly added yaml (`_include/admission-control-filter.yaml`), and also with an existing config (`_include/bandwidth-limit-filter.yaml`), both passed.

Does the validation step exist? Am happy to set that up if not, or maybe I'm missing something? Lmk. Thanks

**Comment by phlax** [comment]
> Does the validation step exist? 

should be automatic - it is tested in 2 ways - how did you try to test it ?

**Comment by hshamji** [comment]
> how did you try to test it ?

`./ci/run_envoy_docker.sh './ci/do_ci.sh docs`

**Comment by phlax** [comment]
the first test (validation) is run with:

```console
# bazel run "${BAZEL_BUILD_OPTIONS[@]}" //configs:example_configs_validation
```

the second - im not sure off hand how it is triggered, but it requires a built binary and is run later in CI - this one starts a server with each of the configs

File: docs/root/configuration/http/http_filters/_include/admission-control-filter.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+static_resources:
+  listeners:
+    - address:
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/_include/admission-control-filter.yaml:3]: one small nit: could you use "least" indent for the yaml lists 

we dont have ci currently to check this but it makes yaml config more readable i think

File: docs/root/configuration/http/http_filters/admission_control_filter.rst [hunk 2]
```diff
@@ -77,38 +80,11 @@ Example Configuration
...
-          - start: 100
-            end:   400
-          - start: 404
-            end:   404
-      grpc_criteria:
-        grpc_success_status:
-          - 0
-          - 1
+.. literalinclude:: _include/admission-control-filter.yaml
+    :language: yaml
```
Comment (phlax) [inline docs/root/configuration/http/http_filters/admission_control_filter.rst:84]: lets add `:linenos:` - we havent been so consistent with how we present these, but i think its helpful to have line numbers

**Review by phlax — CHANGES_REQUESTED** [review]
@hshamji great! thanks for doing this

a couple of nits inline, and i think you need to fix DCO and then should be good to land

File: docs/root/configuration/http/http_filters/_include/admission-control-filter.yaml (hunk 1 above)
Comment (hshamji) [reply docs/root/configuration/http/http_filters/_include/admission-control-filter.yaml]: Done. Didn't see a change with `yq -I1` but pyyaml seemed to do the trick.

**Review by phlax — DISMISSED** [review]
lgtm, thanks @hshamji

**Comment by hshamji** [comment]
Thanks @phlax. Happy to help with any other pipeline/build/CI issues as they arise; or any larger pieces of work too.

**Pull request merged**